### ISSUE TITLE
Duplicate detections triggers on `*Nats-Msg-Id` in the header not on an exact `Nats-Msg-Id` match

### ISSUE THREAD

**Issue by osmanovv** [nats-io/nats-server#3645]
## Defect

I am testing on Windows Server 2019 with the following version of the nats-server:
```
>nats-server -DV
[2288] 2022/11/17 15:18:26.916148 [[32mINF[0m] Starting nats-server
[2288] 2022/11/17 15:18:26.991985 [[32mINF[0m]   Version:  2.9.3
[2288] 2022/11/17 15:18:26.992503 [[32mINF[0m]   Git:      [25e82d7]
[2288] 2022/11/17 15:18:26.992545 [[36mDBG[0m]   Go build: go1.19.2

>nats --version
0.0.34
```

Using JetStream named `ORDERS` with retention: `WorkQueue`.

#### Steps or code to reproduce the issue:

Send three messages: first one with `Nats-Msg-Id` header, second one with `Orig-Nats-Msg-Id` and the third with `Original-Nats-Msg-Id`:
```
nats pub ORDERS.test "test" -H "Nats-Msg-Id:1"
nats pub ORDERS.test "test" -H "Orig-Nats-Msg-Id:1"
nats pub ORDERS.test "test" -H "Original-Nats-Msg-Id:1"
```

#### Expected result:
It must be three different messages.

#### Actual result:
But actually there is only one message in that subject:
```
>nats stream subjects
? Select a Stream ORDERS

      ORDERS.test: 1                     : 0                     : 0
```

Even more, if we publish to different subject, stream keeps only the first message:
```
nats pub ORDERS.test1 "test" -H "Nats-Msg-Id:1"
nats pub ORDERS.test2 "test" -H "Orig-Nats-Msg-Id:1"
nats pub ORDERS.test3 "test" -H "Original-Nats-Msg-Id:1"

>nats stream subjects
? Select a Stream ORDERS

      ORDERS.test1: 1                     : 0                     : 0
```

**Comment by ripienaar** [comment]
Can confirm this bug 

```
$ nats req js.in.orders_0 1 -H "Nats-Msg-Id:1"
12:47:04 Sending request on "js.in.orders_0"
12:47:04 Received with rtt 3.093932ms
{"stream":"ORDERS_0", "domain":"hub", "seq":9021015}

$ nats req js.in.orders_0 1 -H "Orig-Nats-Msg-Id:1"
12:47:11 Sending request on "js.in.orders_0"
12:47:11 Received with rtt 1.244954ms
{"stream":"ORDERS_0", "domain":"hub", "seq":9021015,"duplicate": true}

$ nats req js.in.orders_0 1 -H "Original-Nats-Msg-Id:1"
12:47:20 Sending request on "js.in.orders_0"
12:47:20 Received with rtt 1.150984ms
{"stream":"ORDERS_0", "domain":"hub", "seq":9021015,"duplicate": true}
```

see the `duplicate` warning

**Comment by derekcollison** [comment]
This fix will be in 2.9.7 which we are releasing today, thanks for the report!
